src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/ReviewVisibleEventHandlers.cs

```diff
@@ -10,46 +11,58 @@ namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
-                                                                             isPortedFxCopRule: true,
-                                                                             isDataflowRule: false,
-                                                                             isEnabledByDefaultInFxCopAnalyzers: false);
-        internal static DiagnosticDescriptor DefaultRule = DiagnosticDescriptorHelper.Create(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageDefault,
-                                                                             DiagnosticCategory.Security,
-                                                                             RuleLevel.Disabled,
-                                                                             description: s_localizableDescription,
-                                                                             isPortedFxCopRule: true,
-                                                                             isDataflowRule: false,
-                                                                             isEnabledByDefaultInFxCopAnalyzers: false);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-        //ImmutableArray.Create(SecurityRule, DefaultRule);
-
-#pragma warning disable RS1025 // Configure generated code analysis
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageDefault,
+            DiagnosticCategory.Security,
+            RuleLevel.Disabled,
+            description: s_localizableDescription,
+            isPortedFxCopRule: true,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(context =>
+            {
+                var eventArgsType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+
+                context.RegisterSymbolAction(context =>
+                {
+                    var method = (IMethodSymbol)context.Symbol;
+
+                    if (!method.IsExternallyVisible())
+                    {
+                        return;
+                    }
+
+                    if (!method.HasEventHandlerSignature(eventArgsType))
```

Comment:
Can you update `HasEventHandlerSignature`:
1. Update doc comments that still mention that second parameter type is or derives from `eventArgsType`
2. Remove the `[NotNullWhen(returnValue: true)]` attribute on `eventArgsType` parameter?

https://github.com/dotnet/roslyn-analyzers/blob/2763ec7c3d235319e15c3d5d9db111e1715f9be4/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L561-L566

Reply:
1. I will update the comment to say we also mention the name ends with

2. The first line of the method is still `=> eventArgsType != null &&` so that seems to be fair keep it. Did you meant that you feel like we can get rid of the null check and allow the method to return `true` even when `EventArgs` type was not found? I have some doubts because if we can't find this type in system we are most likely screwed anyways for something else. Let me know your preference.

3. The helper method mentions that the method returns `void` which isn't tested. Shall I update the doc or the method? I would be in favor of updating the implementation to check for `void`.